pkg/k8s/k8s.go

```diff
@@ -162,6 +196,22 @@ func BoolToConditionStatus(val bool) metav1.ConditionStatus {
 	return metav1.ConditionFalse
 }
 
+// ConditionStatusToBool converts condition status to bool.
+func ConditionStatusToBool(status metav1.ConditionStatus) bool {
+	return status == metav1.ConditionTrue
+}
+
+// ResetConditionIfMatches resets the condition values to default if the type, reason and message matches.
+func ResetConditionIfMatches(conditions []metav1.Condition, ctype string, reason, message, newReason string) {
```

Comment:
This function is not needed. Use `UpdateCondition()`

Reply:
This function is required because we need to reset the condition to default only if all the condition value matches. Unlike `UpdateCondition()` where only condition type is checked